# Perfil de Aplicativos Lucrativos

Esta análise tem como objetivo buscar quais os tipos de aplicativos não pagos, dentre aqueles disponíveis na *Google Play Store* e na *App Store* da *Apple* são os mais buscados pelos usuários.

Os aplicativos a serem desenvolvidos serão disponibilizados gratuitamente e a fonte do lucro serão os anúncios veiculados neles. Partindo desta premissa, os aplicativos mais lucrativos são aqueles que atraem mais usuários.
Os resultados obtidos por esta análise serão uma base para a tomada de decisão dos desenvolvedores em relação a qual tipo de aplicativo desenvolver.

## Explorando os dados

Serão utilizados dois diferentes conjuntos de dados, ambos disponibilizados pelo [*Kaggle*](https://www.kaggle.com/) sendo eles:

   
   * [*Google Play Store Apps*](https://www.kaggle.com/lava18/google-play-store-apps) - contém informações relativas a aproximadamente 10 mil diferentes aplicativos disponíveis na *Google Play Store*. Dados coletados em agosto de 2018.
   * [*Mobile App Store (7200 apps)*](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) - contém informações relativas a aproximadamente 7 mil aplicativos disponíveis na *App Store*. Dados coletados em Julho de 2017.

In [1]:
from IPython.display import Markdown, display
from csv import reader

### Google Play Store ###
csv = open('Arquivos/googleplaystore.csv')
read_csv = reader(csv)
apps_android = list(read_csv)
apps_android_header = apps_android[0]
apps_android = apps_android[1:]

### App Store ###
csv = open('Arquivos/AppleStore.csv')
read_csv = reader(csv)
apps_ios = list(read_csv)
apps_ios_header = apps_ios[0]
apps_ios = apps_ios[1:]

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        display(Markdown('#### Informações do *dataset*'))
        print('Quantidade de linhas:', len(dataset))
        print('Quantidade de colunas:', len(dataset[0]))

In [3]:
display(Markdown('#### Colunas do *Play Store*'))
print(apps_android_header)

display(Markdown('#### Dados do *Play Store*'))
explore_data(apps_android, 0, 5, True)

#### Colunas do *Play Store*

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


#### Dados do *Play Store*

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']




#### Informações do *dataset*

Quantidade de linhas: 10841
Quantidade de colunas: 13


In [4]:
display(Markdown('#### Colunas do *App Store*'))
print(apps_ios_header)

display(Markdown('#### Dados do *App Store*'))
explore_data(apps_ios, 0, 5, rows_and_columns=True)

#### Colunas do *App Store*

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


#### Dados do *App Store*

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']




#### Informações do *dataset*

Quantidade de linhas: 7197
Quantidade de colunas: 16


## Limpeza dos dados

Nesta etapa devemos garantir que os dados estejam precisos, eliminando duplicatas e corrigindo dados ausentes.
Além disso, nosso objetivo é considerar apenas os aplicativos que sejam gratuitos e de língua inglesa, portanto devemos deletar do nosso conjunto de dados os aplicativos que não possuam estas características.

### 1. Dados não precisos

Vamos começar a limpeza realizando a correção dos dados não precisos presentes em ambos os conjuntos de dados.

In [5]:
def verify_data_columns(dataset, header):
    tam = len(header)
    print('Quantidade de colunas: '+ str(tam), end='\n\n')
    
    for i in range(len(dataset)):
        if(len(dataset[i]) != tam):
            print('Quantidade diferente de colunas para o índice ' + str(i) + ': ' + str(len(dataset[i])) + ' colunas')
            print(dataset[i], end='\n\n')

In [6]:
verify_data_columns(apps_android, apps_android_header)

Quantidade de colunas: 13

Quantidade diferente de colunas para o índice 10472: 12 colunas
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']



In [7]:
print(apps_android[10471])
print(apps_android[10472])
print(apps_android[10473])

['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


A partir da análise acima, verifica-se que os dados do *Play Store* devem ter informações referentes a 13 atributos. No entanto a linha 10472 possui apenas 12. Comparando-a com as linhas imediatamante acima e abaixo, podemos observar que o índice da coluna faltante é 1. Vamos deletar esta linha com dados faltantes.

In [8]:
del apps_android[10472]

In [9]:
verify_data_columns(apps_ios, apps_ios_header)

Quantidade de colunas: 16



### 2. Remoção de duplicatas

O conjunto de dados *Play Store* também possui duplicatas de alguns aplicativos, apenas mudando a informação relativa à quantidade de *reviews*, como pode-se ver na saída do trecho de código abaixo.

In [10]:
for app in apps_android:
    if app[0] == 'Instagram': print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


Vamos então verificar quantas duplicatas de aplicativos existem para decidir qual é a melhor abordagem para removê-las.

In [11]:
def verify_data_duplicates(dataset):
    
    duplicate_apps = []
    unique_apps = []

    for app in dataset:
        name = app[0]
        if name in unique_apps:
            duplicate_apps.append(name)
        else:
            unique_apps.append(name)

    print('Quantidade de aplicativos duplicados:', len(duplicate_apps), end='\n\n')
    print('Exemplos de aplicativos duplicados:', duplicate_apps[:15])

In [12]:
verify_data_duplicates(apps_android)

Quantidade de aplicativos duplicados: 1181

Exemplos de aplicativos duplicados: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


In [13]:
verify_data_duplicates(apps_ios)

Quantidade de aplicativos duplicados: 0

Exemplos de aplicativos duplicados: []


Como verificado acima, apenas o conjunto de dados da *Play Store* possui as duplicatas, o da *App Store* não as possui. A estratégia adotada será a de utilizar a linha que possui maior quantidade de *reviews* (informação no index 3, indica que aquele dado foi recolhido mais recentemente) e as restantes serão eliminadas.

In [14]:
dict_unique_android = {}

for app in apps_android:
    name = app[0]
    n_reviews = float(app[3])
    
    if(name not in dict_unique_android.keys()):
        dict_unique_android[name] = n_reviews
    else:
        dict_unique_android[name] = dict_unique_android[name] if dict_unique_android[name] > n_reviews else n_reviews

In [15]:
apps_android_clean = []
apps_added = []

for app in apps_android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (n_reviews == dict_unique_android[name]) and (name not in apps_added):
        apps_added.append(name)
        apps_android_clean.append(app)

In [16]:
print('Quantidade de apps no conjunto original - quantidade de duplicatas:' , len(apps_android)-1181)
print('Quantidade de apps no novo conjunto:' , len(apps_android_clean))

Quantidade de apps no conjunto original - quantidade de duplicatas: 9659
Quantidade de apps no novo conjunto: 9659


In [17]:
explore_data(apps_android_clean, 0, 0, True)

#### Informações do *dataset*

Quantidade de linhas: 9659
Quantidade de colunas: 13


### 3. Separação dos aplicativos desenvolvidos em inglês

Como mencionado anteriormente, o público alvo para esta análise é aquele falante da língua inglesa. Se observarmos o nome de alguns dos aplicativos podemos observar claramente que não foram desenvolvidos em inglês:


In [18]:
display(Markdown('#### Apps da *Play Store*'))
print(apps_android_clean[3168][0])
print(apps_android_clean[5205][0])
print(apps_android_clean[8131][0])
print(apps_android_clean[9582][0])
print()
display(Markdown('#### Apps da *App Store*'))
print(apps_ios[1427][1])
print(apps_ios[2512][1])
print(apps_ios[6675][1])
print(apps_ios[6831][1])

#### Apps da *Play Store*

صور حرف H
뽕티비 - 개인방송, 인터넷방송, BJ방송
أحداث وحقائق | خبر عاجل في اخبار العالم
FP Разбитый дисплей



#### Apps da *App Store*

优酷视频
Dictionary ( قاموس عربي / انجليزي + ودجيت الترجمة)
スーパーストリートファイターIV パチスロエディション
Übungen für einen starken Rücken – GEO WISSEN GESUNDHEIT


Vamos então utilizar uma função para identificar todos aqueles nomes de aplicativos que possuem mais de 3 caracteres diferentes de 0 e 127 na tabela ASCII (que são comumente os caracteres utilizados na língua inglesa).

In [19]:
def non_english_char(name):
    if sum([1 if ord(c) > 172 else 0 for c in name]) > 3: return True
    else: return False

In [20]:
print(non_english_char('Instagram'))
print(non_english_char('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(non_english_char('Docs To Go™ Free Office Suite'))
print(non_english_char('Instachat 😜'))

False
True
False
False


In [21]:
apps_android_eng = []

for app in apps_android_clean:
    if not non_english_char(app[0]): apps_android_eng.append(app)
        
explore_data(apps_android_eng, 0, 0, True)

#### Informações do *dataset*

Quantidade de linhas: 9614
Quantidade de colunas: 13


In [22]:
apps_ios_eng = []

for app in apps_ios:
    if not non_english_char(app[1]): apps_ios_eng.append(app)
        
explore_data(apps_ios_eng, 0, 0, True)

#### Informações do *dataset*

Quantidade de linhas: 6183
Quantidade de colunas: 16


### 4. Separação dos aplicativos gratuitos

O aplicativo a ser desenvolvido será gratuito, e a fonte de lucro serão os anúncios. Então para obter uma análise mais assertiva, é necessário considerar apenas os aplicativos gratuitos. A próxima etapa então será manter apenas os aplicativos gratuitos.


In [23]:
apps_android_free = []
non_free = []

for app in apps_android_eng:
    if app[7] == '0': apps_android_free.append(app)
    else: non_free.append(app[7])

prices = [float(p.replace('$','')) for p in list(set(non_free))]
print('Checar se não ficou algum de fora:', sorted(prices))
explore_data(apps_android_free, 0, 0, True)

Checar se não ficou algum de fora: [0.99, 1.0, 1.04, 1.2, 1.26, 1.29, 1.49, 1.5, 1.59, 1.61, 1.7, 1.75, 1.76, 1.96, 1.97, 1.99, 2.0, 2.49, 2.5, 2.56, 2.59, 2.6, 2.9, 2.95, 2.99, 3.02, 3.04, 3.08, 3.28, 3.49, 3.61, 3.88, 3.9, 3.95, 3.99, 4.29, 4.49, 4.59, 4.6, 4.77, 4.8, 4.84, 4.85, 4.99, 5.0, 5.49, 5.99, 6.49, 6.99, 7.49, 7.99, 8.49, 8.99, 9.0, 9.99, 10.0, 10.99, 11.99, 12.99, 13.99, 14.0, 14.99, 15.46, 15.99, 16.99, 17.99, 18.99, 19.4, 19.9, 19.99, 24.99, 25.99, 28.99, 29.99, 30.99, 33.99, 37.99, 39.99, 46.99, 74.99, 79.99, 89.99, 109.99, 154.99, 200.0, 299.99, 379.99, 389.99, 394.99, 399.99, 400.0]


#### Informações do *dataset*

Quantidade de linhas: 8864
Quantidade de colunas: 13


In [24]:
apps_ios_free = []
non_free = []

for app in apps_ios_eng:
    if app[4] == '0.0' or app[4] == '0': apps_ios_free.append(app)
    else: non_free.append(app[4])


prices = [p.replace('$','') for p in list(set(non_free))]
print('Checar se não ficou algum de fora:', sorted(prices))
explore_data(apps_ios_free, 0, 0, True)

Checar se não ficou algum de fora: ['0.99', '1.99', '11.99', '12.99', '13.99', '14.99', '15.99', '16.99', '17.99', '18.99', '19.99', '2.99', '20.99', '21.99', '22.99', '24.99', '249.99', '27.99', '29.99', '299.99', '3.99', '34.99', '39.99', '4.99', '49.99', '5.99', '59.99', '6.99', '7.99', '74.99', '8.99', '9.99', '99.99']


#### Informações do *dataset*

Quantidade de linhas: 3222
Quantidade de colunas: 16


## Análise dos dados

Uma vez que os dados foram preparados através da limpeza, é possível então iniciar a análise.
Deve-se ter em mente que estamos buscando os aplicativos mais populares, uma vez que a receita obtida será altamente influenciada pela quantidade de pessoas usando o aplicativo que será desenvolvido.

A estratégia a ser adotada será a seguinte:
1. Desenvolver uma versão mínima do aplicativo para *Android* e adicioná-la à *Google Play Store*;
2. Desenvolver além da versão mínima, se a recepção pelos usuários for boa;
3. Desenvolver uma versão *iOS* e adicioná-la à *App Store*, se após seis meses a versão para *Android* apresentar uma boa lucratividade.

Como o objetivo final é disponibilizar o aplicativo para ambos sistemas operacionais, é necessário encontrar um perfil de aplicativo que sirva para ambos os *markets*.

#### Categorias e gêneros mais comuns

Vamos começar buscando os gêneros de aplicativos mais comuns.

In [25]:
def freq_table(dataset, index):
    freq = {}
    freq_perc = {}
    total = 0
    
    for app in dataset:
        total += 1
        if(app[index] not in freq):
            freq[app[index]] = 1
        else:
            freq[app[index]] += 1
            
    for key in freq:
        freq_perc[key] = round((freq[key] / total) * 100, 2)
    
    return freq_perc
    

In [26]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1] + ': ' + str(entry[0]) + '%')

In [27]:
display(Markdown('#### *Play Store*: *Category*'))
display_table(apps_android_free, 1)

#### *Play Store*: *Category*

FAMILY: 18.91%
GAME: 9.72%
TOOLS: 8.46%
BUSINESS: 4.59%
LIFESTYLE: 3.9%
PRODUCTIVITY: 3.89%
FINANCE: 3.7%
MEDICAL: 3.53%
SPORTS: 3.4%
PERSONALIZATION: 3.32%
COMMUNICATION: 3.24%
HEALTH_AND_FITNESS: 3.08%
PHOTOGRAPHY: 2.94%
NEWS_AND_MAGAZINES: 2.8%
SOCIAL: 2.66%
TRAVEL_AND_LOCAL: 2.34%
SHOPPING: 2.25%
BOOKS_AND_REFERENCE: 2.14%
DATING: 1.86%
VIDEO_PLAYERS: 1.79%
MAPS_AND_NAVIGATION: 1.4%
FOOD_AND_DRINK: 1.24%
EDUCATION: 1.16%
ENTERTAINMENT: 0.96%
LIBRARIES_AND_DEMO: 0.94%
AUTO_AND_VEHICLES: 0.93%
HOUSE_AND_HOME: 0.82%
WEATHER: 0.8%
EVENTS: 0.71%
PARENTING: 0.65%
ART_AND_DESIGN: 0.64%
COMICS: 0.62%
BEAUTY: 0.6%


Para a amostra de aplicativos gratuitos de língua inglesa do *Android*, é possível visualizar que a maior quantidade é referente à categoria **Família**, somando aproximadamente o dobro (18,91% dos aplicativos) da segunda categoria com a maior quantidade, que é a de **Jogo** (9,72% dos aplicativos). Em terceiro vem a categoria **Ferramentas**, com 8,46% dos aplicativos.

Em uma análise mais geral, pode-se afirmar que há uma quantia significativa de aplicativos com fins práticos (como família, ferramentas, negócios, estilo de vida e produtividade, por exemplo) e que nenhuma categoria se destaca muito em quantidade.

In [28]:
display(Markdown('#### *Play Store*: *Genres*'))
display_table(apps_android_free, 9)

#### *Play Store*: *Genres*

Tools: 8.45%
Entertainment: 6.07%
Education: 5.35%
Business: 4.59%
Productivity: 3.89%
Lifestyle: 3.89%
Finance: 3.7%
Medical: 3.53%
Sports: 3.46%
Personalization: 3.32%
Communication: 3.24%
Action: 3.1%
Health & Fitness: 3.08%
Photography: 2.94%
News & Magazines: 2.8%
Social: 2.66%
Travel & Local: 2.32%
Shopping: 2.25%
Books & Reference: 2.14%
Simulation: 2.04%
Dating: 1.86%
Arcade: 1.85%
Video Players & Editors: 1.77%
Casual: 1.76%
Maps & Navigation: 1.4%
Food & Drink: 1.24%
Puzzle: 1.13%
Racing: 0.99%
Role Playing: 0.94%
Libraries & Demo: 0.94%
Auto & Vehicles: 0.93%
Strategy: 0.91%
House & Home: 0.82%
Weather: 0.8%
Events: 0.71%
Adventure: 0.68%
Comics: 0.61%
Beauty: 0.6%
Art & Design: 0.6%
Parenting: 0.5%
Card: 0.45%
Casino: 0.43%
Trivia: 0.42%
Educational;Education: 0.39%
Board: 0.38%
Educational: 0.37%
Education;Education: 0.34%
Word: 0.26%
Casual;Pretend Play: 0.24%
Music: 0.2%
Racing;Action & Adventure: 0.17%
Puzzle;Brain Games: 0.17%
Entertainment;Music & Video: 0.17%
Casual;

Em relação aos gêneros para o *Android*, o primeiro colocado em quantidade é **Ferramentas**, o segundo **Entretenimento** e o terceiro **Educação**. Estes resultados condizem com as categorias mais frequentes, apesar da falta de uma relação muito clara entre categoria e gênero.
De uma maneira geral, para os aplicativos do *Android*, a maior quantidade é de aplicativos de ferramentas.

In [29]:
display(Markdown('#### *App Store*: *prime_genre*'))
display_table(apps_ios_free, 11)

#### *App Store*: *prime_genre*

Games: 58.16%
Entertainment: 7.88%
Photo & Video: 4.97%
Education: 3.66%
Social Networking: 3.29%
Shopping: 2.61%
Utilities: 2.51%
Sports: 2.14%
Music: 2.05%
Health & Fitness: 2.02%
Productivity: 1.74%
Lifestyle: 1.58%
News: 1.33%
Travel: 1.24%
Finance: 1.12%
Weather: 0.87%
Food & Drink: 0.81%
Reference: 0.56%
Business: 0.53%
Book: 0.43%
Navigation: 0.19%
Medical: 0.19%
Catalogs: 0.12%


Para a amostra com as mesmas características do *iOS* a maior quantidade de aplicativos é do gênero **Jogos**, somando 58,16% dos aplicativos, com muita diferença para a segundo maior, **Entretenimento**, que possui menos de 8% dos aplicativos.

Observando os gêneros de aplicativos mais frequentes na amostra, é possível afimar que para o *iOS* os aplicativos de entretenimento (jogos, redes sociais, foto & vídeo) são os de maior quantidade para a *App Store*, de maneira que os aplicativos com fins práticos (dentre vários outros, saúde, produtividade, utilidades e clima) são uma quantidade menor, o contrário do *Android*.

A análise relativa aos aplicativos mais comuns nas duas *stores* mostrou que os aplicativos para *iOS* são predominantemente jogos, enquanto o mesmo não pode ser observado para o *Android*, que apresenta uma distribuição mais equilibrada entre jogos e aplicativos para fins práticos.

No entanto esta análise em si não diz muito sobre a popularidade dos aplicativos, ela apenas oferece informações sobre a disponibilidade dos aplicativos, não sendo capaz de oferecer *insights* sobre a demanda. Então vamos tentar utilizar as informações de ambos os conjuntos de dados para descobrir mais sobre a popularidade dos aplicativos.

### Gêneros de aplicativos mais populares

O objetivo é descobrir quais são os aplicativos mais populares nas amostras que temos em mãos. Para o *Android* vamos utilizar a quantidade de downloads (da coluna `Installs`, de índice 5). Como para *iOS* não existe essa informação, utilizaremos a quantidade de avaliações (coluna `rating_count_tot`, de índice 5).

#### *App Store*

Como já mencionado, aqui não temos como para o conjunto do *Android* uma quantidade aproximada de instalações, mas sim uma quantidade de avaliações, que é geralmente bem menor do que a primeira. Além disso, não temos padronizado o que seria uma alta quantidade de avaliações para todos os aplicativos. Por este motivo, não vamos adotar o mesmo critério de remoção de aplicativos com muitas avaliações.

In [30]:
display(Markdown('#### Número médio de avaliações por gênero'))

ios_genres = freq_table(apps_ios_free, 11).keys()

for genre in ios_genres:
    total = 0
    len_genre = 0
    
    for app in apps_ios_free:
        app_genre = app[11]
        if app_genre == genre:
            rating_count = float(app[5])
            total += rating_count
            len_genre += 1
    
    print(genre + ': ' + str(round(total / len_genre, 2)))


#### Número médio de avaliações por gênero

Social Networking: 71548.35
Photo & Video: 28441.54
Games: 22788.67
Music: 57326.53
Reference: 74942.11
Health & Fitness: 23298.02
Weather: 52279.89
Utilities: 18684.46
Travel: 28243.8
Shopping: 26919.69
News: 21248.02
Navigation: 86090.33
Lifestyle: 16485.76
Entertainment: 14029.83
Food & Drink: 33333.92
Sports: 23008.9
Book: 39758.5
Finance: 31467.94
Education: 7003.98
Productivity: 21028.41
Business: 7491.12
Catalogs: 4004.0
Medical: 612.0


Obtidas as médias, vamos analisar os gêneros que possuem as maiores, sendo eles:

1. Navigation
2. Reference
3. Social Networking
4. Music
5. Weather
6. Book
7. Food & Drink
8. Finance

In [31]:
for app in apps_ios_free:
    if(app[11] == 'Navigation'): print(app[1] + ': ' + app[5])

Waze - GPS Navigation, Maps & Real-time Traffic: 345046
Google Maps - Navigation & Transit: 154911
Geocaching®: 12811
CoPilot GPS – Car Navigation & Offline Maps: 3582
ImmobilienScout24: Real Estate Search in Germany: 187
Railway Route Search: 5


Aqui pode-se observar que *Waze* e *Google Maps* são os grandes aplicativos que puxam a média pra cima, de modo que a quantidade de avaliações neste gênero não é bem distribuída. Além disso, a quantidade de aplicativos deste gênero é baixa. Em outras palavras, a popularidade dos dois aplicativos mencionados é muito grande, o que mascara a real popularidade de todos os outros aplicativos do mesmo gênero.

In [32]:
for app in apps_ios_free:
    if(app[11] == 'Reference'): print(app[1] + ': ' + app[5]) 

Bible: 985920
Dictionary.com Dictionary & Thesaurus: 200047
Dictionary.com Dictionary & Thesaurus for iPad: 54175
Google Translate: 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran: 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition: 17588
Merriam-Webster Dictionary: 16849
Night Sky: 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE): 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools: 4693
GUNS MODS for Minecraft PC Edition - Mods Tools: 1497
Guides for Pokémon GO - Pokemon GO News and Cheats: 826
WWDC: 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free: 718
VPN Express: 14
Real Bike Traffic Rider Virtual Reality Glasses: 8
教えて!goo: 0
Jishokun-Japanese English Dictionary & Translator: 0


O mesmo pode se observar para o gênero `Reference`, no qual o aplicativo *Bible* e o *Dictionary.com Dictionary & Thesaurus* juntos tem uma quantidade enorme de avaliações em comparação com o restante dos aplicativos.

Há potencial na indicação deste gênero. Se apenas os aplicativos para *iOS* estivessem sendo considerados, esta seria uma provável indicação.

In [33]:
for app in apps_ios_free:
    if(app[11] == 'Social Networking'): print(app[1] + ': ' + app[5])

Facebook: 2974676
Pinterest: 1061624
Skype for iPhone: 373519
Messenger: 351466
Tumblr: 334293
WhatsApp Messenger: 287589
Kik: 260965
ooVoo – Free Video Call, Text and Voice: 177501
TextNow - Unlimited Text + Calls: 164963
Viber Messenger – Text & Call: 164249
Followers - Social Analytics For Instagram: 112778
MeetMe - Chat and Meet New People: 97072
We Heart It - Fashion, wallpapers, quotes, tattoos: 90414
InsTrack for Instagram - Analytics Plus More: 85535
Tango - Free Video Call, Voice and Chat: 75412
LinkedIn: 71856
Match™ - #1 Dating App.: 60659
Skype for iPad: 60163
POF - Best Dating App for Conversations: 52642
Timehop: 49510
Find My Family, Friends & iPhone - Life360 Locator: 43877
Whisper - Share, Express, Meet: 39819
Hangouts: 36404
LINE PLAY - Your Avatar World: 34677
WeChat: 34584
Badoo - Meet New People, Chat, Socialize.: 34428
Followers + for Instagram - Follower Analytics: 28633
GroupMe: 28260
Marco Polo Video Walkie Talkie: 27662
Miitomo: 23965
SimSimi: 23530
Grindr - G

Para `Social Networking`, *Facebook* e *Pinterest* somam quase 4 milhões de avaliações, enquanto o terceiro aplicativo com mais avaliações é o *Skype for iPhone* com 373,5 mil. O nicho das redes sociais já é muito dominado por gigantes, o que tornaria um aplicativo novo muito pouco competitivo a não ser que se investisse muito nele. Então o gênero `Social Networking` está descartado.

In [34]:
for app in apps_ios_free:
    if(app[11] == 'Music'): print(app[1] + ': ' + app[5])

Pandora - Music & Radio: 1126879
Spotify Music: 878563
Shazam - Discover music, artists, videos & lyrics: 402925
iHeartRadio – Free Music & Radio Stations: 293228
SoundCloud - Music & Audio: 135744
Magic Piano by Smule: 131695
Smule Sing!: 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music: 110420
Amazon Music: 106235
SoundHound Song Search & Music Player: 82602
Sonos Controller: 48905
Bandsintown Concerts: 30845
Karaoke - Sing Karaoke, Unlimited Songs!: 28606
My Mixtapez Music: 26286
Sing Karaoke Songs Unlimited with StarMaker: 26227
Ringtones for iPhone & Ringtone Maker: 25403
Musi - Unlimited Music For YouTube: 25193
AutoRap by Smule: 18202
Spinrilla - Mixtapes For Free: 15053
Napster - Top Music & Radio: 14268
edjing Mix:DJ turntable to remix and scratch music: 13580
Free Music - MP3 Streamer & Playlist Manager Pro: 13443
Free Piano app by Yokee: 13016
Google Play Music: 10118
Certified Mixtapes - Hip Hop Albums & Mixtapes: 9975
TIDAL: 7398
YouTube Music: 7109
Nicki Minaj: The

Do mesmo modo que para os anteriores, em `Music` alguns aplicativos muito populares de empresas grandes dominam a quantiade das avaliações na média, e não seria interessante criar um novo aplicativo que não teria muita competitividade no mercado. Por este motivo, a categoria `Music` também está descartada.

As próximas opções seriam `Weather`, `Book`, `Food & Drink` e `Finance`. No entanto, há algumas limitações em relação a elas ao se observar o escopo do aplicativo que deseja-se desenvolver.

* `Weather`: a não ser que desenvolva-se algo muito inovador no segmento de clima que prenda a atenção dos usuários (como por exemplo a junção de um *planner* com informações climáticas), este gênero não é interessante, uma vez que usuários passam pouco tempo utilizando os aplicativos de clima. Além disso, é possível que tivesse de se pagar por informações de uma API;
* `Food & Drink`: *Domino's*, *Starbucks* e *McDonald's* são empresas que com grandes quantidades de avaliações para este gênero. Como não se deseja criar de fato uma cozinha para produção de alimentos, vamos tambémd descartar este gênero;
* `Finance`: este poderia ser uma boa indicação em relação a tempo do usuário no aplicativo, a não ser pelo fato de que um especialista sobre finanças teria de ser contratado para o desenvolvimento do aplicativo, e esta não é a intenção.

Portanto a partir da análise da amostra dos aplicativos do *App Store* mantêm-se dois gêneros possíveis: o de `Reference` e `Books`. Nesta linha, poderia ser desenvolvido um aplicativo que disponibilizasse livros de domínio público, oferecendo a opção de *audiobooks*, além de criar fóruns de discussões para que os usuários pudessem compartilhar suas impressões.

Vamos agora verificar quais são os gêneros mais populares para a *Play Store*, e se este tipo de indicação funcionaria também para o *Android*.

#### *Play Store*

A coluna `Installs` não possui números precisos em relação à quantidade de instalações realizadas para aquele determinado aplicativo, no entanto não é necessária uma grande precisão, apenas importa qual é a relação entre os diferentes aĺicativos. Neste caso, um aplicativo que tenha `1,000,000+` possui mais *downloads* do que aquele que tenha `10,000+`.

In [35]:
display_table(apps_android_free, 5)

1,000,000+: 15.73%
100,000+: 11.55%
10,000,000+: 10.55%
10,000+: 10.2%
1,000+: 8.39%
100+: 6.92%
5,000,000+: 6.83%
500,000+: 5.56%
50,000+: 4.77%
5,000+: 4.51%
10+: 3.54%
500+: 3.25%
50,000,000+: 2.3%
100,000,000+: 2.13%
50+: 1.92%
5+: 0.79%
1+: 0.51%
500,000,000+: 0.27%
1,000,000,000+: 0.23%
0+: 0.05%
0: 0.01%


Vamos então calcular a média de `Installs` por categoria.

In [36]:
android_categories = freq_table(apps_android_free, 1).keys()

for category in android_categories:
    total = 0
    len_category = 0
    
    for app in apps_android_free:
        category_app = app[1]
        if category_app == category and (app[5] != '1,000,000,000+' and app[5] != '500,000,000+' and app[5] != '100,000,000+'):
            n_installs = float(app[5].replace('+','').replace(',',''))
            total += n_installs
            len_category += 1
            
    print(category + ': ' + str(round(total / len_category, 2)))
    

ART_AND_DESIGN: 1986335.09
AUTO_AND_VEHICLES: 647317.82
BEAUTY: 513151.89
BOOKS_AND_REFERENCE: 1437212.22
BUSINESS: 1226918.74
COMICS: 817657.27
COMMUNICATION: 3603485.39
DATING: 854028.83
EDUCATION: 1833495.15
ENTERTAINMENT: 6118250.0
EVENTS: 253542.22
FINANCE: 1086125.79
FOOD_AND_DRINK: 1924897.74
HEALTH_AND_FITNESS: 2005713.66
HOUSE_AND_HOME: 1331540.56
LIBRARIES_AND_DEMO: 638503.73
LIFESTYLE: 1152128.78
GAME: 6272564.69
FAMILY: 2342897.53
MEDICAL: 120550.62
SOCIAL: 3084582.52
SHOPPING: 4640920.54
PHOTOGRAPHY: 7670532.29
SPORTS: 2994082.55
TRAVEL_AND_LOCAL: 2944079.63
TOOLS: 3191461.13
PERSONALIZATION: 2549775.83
PRODUCTIVITY: 3379657.32
PARENTING: 542603.62
WEATHER: 5074486.2
VIDEO_PLAYERS: 5544878.13
NEWS_AND_MAGAZINES: 1502841.88
MAPS_AND_NAVIGATION: 2484104.75


As categorias de aplicativos que possuem uma maior média de *downloads* e instalações são:

1. COMMUNICATION
2. VIDEO_PLAYERS
3. SOCIAL
4. PHOTOGRAPHY
5. PRODUCTIVITY

Vamos então observar mais de perto a quantidade de instalações para as categorias no Top 5 de maior média.

In [37]:
for app in apps_android_free:
    if(app[1] == 'COMMUNICATION'): print(app[0] + ': ' + app[5])

WhatsApp Messenger: 1,000,000,000+
Messenger for SMS: 10,000,000+
My Tele2: 5,000,000+
imo beta free calls and text: 100,000,000+
Contacts: 50,000,000+
Call Free – Free Call: 5,000,000+
Web Browser & Explorer: 5,000,000+
Browser 4G: 10,000,000+
MegaFon Dashboard: 10,000,000+
ZenUI Dialer & Contacts: 10,000,000+
Cricket Visual Voicemail: 10,000,000+
TracFone My Account: 1,000,000+
Xperia Link™: 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard: 10,000,000+
Skype Lite - Free Video Call & Chat: 5,000,000+
My magenta: 1,000,000+
Android Messages: 100,000,000+
Google Duo - High Quality Video Calls: 500,000,000+
Seznam.cz: 1,000,000+
Antillean Gold Telegram (original version): 100,000+
AT&T Visual Voicemail: 10,000,000+
GMX Mail: 10,000,000+
Omlet Chat: 10,000,000+
My Vodacom SA: 5,000,000+
Microsoft Edge: 5,000,000+
Messenger – Text and Video Chat for Free: 1,000,000,000+
imo free video calls and chat: 500,000,000+
Calls & Text by Mo+: 5,000,000+
free video calls and chat: 50,000

Para a categoria `COMMUNICATION` podemos observar que a média foi puxada para cima por conta de gigantes como *Whatsapp*, *Gmail* e *Skype*, que dominam o mercado.

In [38]:
for app in apps_android_free:
    if(app[1] == 'VIDEO_PLAYERS'): print(app[0] + ': ' + app[5])

YouTube: 1,000,000,000+
All Video Downloader 2018: 1,000,000+
Video Downloader: 10,000,000+
HD Video Player: 1,000,000+
Iqiyi (for tablet): 1,000,000+
Video Player All Format: 10,000,000+
Motorola Gallery: 100,000,000+
Free TV series: 100,000+
Video Player All Format for Android: 500,000+
VLC for Android: 100,000,000+
Code: 10,000,000+
Vote for: 50,000,000+
XX HD Video downloader-Free Video Downloader: 1,000,000+
OBJECTIVE: 1,000,000+
Music - Mp3 Player: 10,000,000+
HD Movie Video Player: 1,000,000+
YouCut - Video Editor & Video Maker, No Watermark: 5,000,000+
Video Editor,Crop Video,Movie Video,Music,Effects: 1,000,000+
YouTube Studio: 10,000,000+
video player for android: 10,000,000+
Vigo Video: 50,000,000+
Google Play Movies & TV: 1,000,000,000+
HTC Service － DLNA: 10,000,000+
VPlayer: 1,000,000+
MiniMovie - Free Video and Slideshow Editor: 50,000,000+
Samsung Video Library: 50,000,000+
OnePlus Gallery: 1,000,000+
LIKE – Magic Video Maker & Community: 50,000,000+
HTC Service—Video P

Podemos observar o mesmo para a categoria de `VIDEO_PLAYERS`, que é dominada pelo *YouTube* e pelo *Google Play Movies*.

In [39]:
for app in apps_android_free:
    if(app[1] == 'SOCIAL'): print(app[0] + ': ' + app[5])

Facebook: 1,000,000,000+
Facebook Lite: 500,000,000+
Tumblr: 100,000,000+
Social network all in one 2018: 100,000+
Pinterest: 100,000,000+
TextNow - free text + calls: 10,000,000+
Google+: 1,000,000,000+
The Messenger App: 1,000,000+
Messenger Pro: 1,000,000+
Free Messages, Video, Chat,Text for Messenger Plus: 1,000,000+
Telegram X: 5,000,000+
The Video Messenger App: 100,000+
Jodel - The Hyperlocal App: 1,000,000+
Hide Something - Photo, Video: 5,000,000+
Love Sticker: 1,000,000+
Web Browser & Fast Explorer: 5,000,000+
LiveMe - Video chat, new friends, and make money: 10,000,000+
VidStatus app - Status Videos & Status Downloader: 5,000,000+
Love Images: 1,000,000+
Web Browser ( Fast & Secure Web Explorer): 500,000+
SPARK - Live random video chat & meet new people: 5,000,000+
Golden telegram: 50,000+
Facebook Local: 1,000,000+
Meet – Talk to Strangers Using Random Video Chat: 5,000,000+
MobilePatrol Public Safety App: 1,000,000+
💘 WhatsLov: Smileys of love, stickers and GIF: 1,000,000+

Para a categoria `SOCIAL`, temos *Facebook*, *Instagram* e *Google+* dominando as instalações.

In [40]:
for app in apps_android_free:
    if(app[1] == 'PHOTOGRAPHY'): print(app[0] + ': ' + app[5])

TouchNote: Cards & Gifts: 1,000,000+
FreePrints – Free Photos Delivered: 1,000,000+
Groovebook Photo Books & Gifts: 500,000+
Moony Lab - Print Photos, Books & Magnets ™: 50,000+
LALALAB prints your photos, photobooks and magnets: 1,000,000+
Snapfish: 1,000,000+
Motorola Camera: 50,000,000+
HD Camera - Best Cam with filters & panorama: 5,000,000+
LightX Photo Editor & Photo Effects: 10,000,000+
Sweet Snap - live filter, Selfie photo edit: 10,000,000+
HD Camera - Quick Snap Photo & Video: 1,000,000+
B612 - Beauty & Filter Camera: 100,000,000+
Waterfall Photo Frames: 1,000,000+
Photo frame: 100,000+
Huji Cam: 5,000,000+
Unicorn Photo: 1,000,000+
HD Camera: 5,000,000+
Makeup Editor -Beauty Photo Editor & Selfie Camera: 1,000,000+
Makeup Photo Editor: Makeup Camera & Makeup Editor: 1,000,000+
Moto Photo Editor: 5,000,000+
InstaBeauty -Makeup Selfie Cam: 50,000,000+
Garden Photo Frames - Garden Photo Editor: 500,000+
Photo Frame: 10,000,000+
Selfie Camera - Photo Editor & Filter & Sticker: 5

Em `PHOTOGRAPHY`, há apenas um que domina as instalações com `1.000.000.000+`, que é o *Google Photos*, e dezoito outros (de edição de fotos ou câmeras com filtros, principalmente) que possuem `100.000.000+`. Se considerássemos apenas o *Android*, esta seria uma categoria interessante de se buscar ter ideias de desenvolvimento de aplicativos. Sigamos com a análise.

In [41]:
for app in apps_android_free:
    if(app[1] == 'PRODUCTIVITY'): print(app[0] + ': ' + app[5])

Microsoft Word: 500,000,000+
All-In-One Toolbox: Cleaner, Booster, App Manager: 10,000,000+
AVG Cleaner – Speed, Battery & Memory Booster: 10,000,000+
QR Scanner & Barcode Scanner 2018: 10,000,000+
Chrome Beta: 10,000,000+
Microsoft Outlook: 100,000,000+
Google PDF Viewer: 10,000,000+
My Claro Peru: 5,000,000+
Power Booster - Junk Cleaner & CPU Cooler & Boost: 1,000,000+
Google Assistant: 10,000,000+
Microsoft OneDrive: 100,000,000+
Calculator - unit converter: 50,000,000+
Microsoft OneNote: 100,000,000+
Metro name iD: 10,000,000+
Google Keep: 100,000,000+
Archos File Manager: 5,000,000+
ES File Explorer File Manager: 100,000,000+
ASUS SuperNote: 10,000,000+
HTC File Manager: 10,000,000+
MyMTN: 1,000,000+
Dropbox: 500,000,000+
ASUS Quick Memo: 10,000,000+
HTC Calendar: 10,000,000+
Google Docs: 100,000,000+
ASUS Calling Screen: 10,000,000+
lifebox: 5,000,000+
Yandex.Disk: 5,000,000+
Content Transfer: 5,000,000+
HTC Mail: 10,000,000+
Advanced Task Killer: 50,000,000+
MyVodafone (India) -

Mais uma vez o *Google* domina uma categoria sozinho, com o *Google Drive*. A categoria de `PRODUCTIVITY` aparenta, pelos nomes, ter uma variabilidade maior em relação à função do aplicativo.

No entanto, assim como todas as categorias do Top 5 maiores médias de instalações analisadas anteriormente, os aplicativos que dominam os *downloads* são de empresas gigantes com as quais é difícil de competir.
Vamos então remover da nossa média os aplicativos que possuem muitos *downloads* (acima de `100.000.000`) pois, como pudemos observar, eles são geralmente hegemônicos em sua categoria.

Como já mencionado, se fossem apenas considerados os aplicativos para *Android*, uma boa indicação seria aplicativos de edição de imagens ou câmeras com filtros, no entanto a mesma popularidade não é observada entre usuários de *iOS*. 

Vamos então verificar se a categoria `BOOKS_AND_REFERENCE` (que foi a indicada para o conjunto dos aplicativos *iOS*) apresentaria uma popularidade considerável entre os usuários de *Android*. Vamos desconsiderar uma quantidade muito grandes de `Installs` - uma vez que de acordo com as análises anteriores, uma quantidade muito grande significa que o aplicativo é de uma gigante -, e também aqueles que tem uma quantidade pequena de *downloads* e instalações - o que indica uma baixa popularidade -, e denominaremos estes aplicativos como "aplicativos médios".



In [42]:
total = 0
qty = 0

for app in apps_android_free:
    if(app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+')):
        print(app[0] + ': ' + app[5])
        total += float(app[5].replace('+','').replace(',',''))
        qty += 1

display(Markdown('#### Média de instalações dos aplicativos médios da categoria `BOOKS_AND_REFERENCE`'))
print(round((total/qty),2))
        

Wikipedia: 10,000,000+
Cool Reader: 10,000,000+
Book store: 1,000,000+
FBReader: Favorite Book Reader: 10,000,000+
Free Books - Spirit Fanfiction and Stories: 1,000,000+
AlReader -any text book reader: 5,000,000+
FamilySearch Tree: 1,000,000+
Cloud of Books: 1,000,000+
ReadEra – free ebook reader: 1,000,000+
Ebook Reader: 5,000,000+
Read books online: 5,000,000+
eBoox: book reader fb2 epub zip: 1,000,000+
All Maths Formulas: 1,000,000+
Ancestry: 5,000,000+
HTC Help: 10,000,000+
Moon+ Reader: 10,000,000+
English-Myanmar Dictionary: 1,000,000+
Golden Dictionary (EN-AR): 1,000,000+
All Language Translator Free: 1,000,000+
Aldiko Book Reader: 10,000,000+
Dictionary - WordWeb: 5,000,000+
50000 Free eBooks & Free AudioBooks: 5,000,000+
Al-Quran (Free): 10,000,000+
Al Quran Indonesia: 10,000,000+
Al'Quran Bahasa Indonesia: 10,000,000+
Al Quran Al karim: 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline: 5,000,000+
Koran Read &MP3 30 Juz Offline: 1,000,000+
Hafizi Quran 15 lines per pag

#### Média de instalações dos aplicativos médios da categoria `BOOKS_AND_REFERENCE`

5312500.0


Em uma primeira olhada, é possível ver que a quantidade de aplicativos é grande (corroborado pela análise anterior, onde constava que do conjunto total, 2.14% são pertencentes a esta categoria). Além disso, a média da quantidade de instalações também é grande, passando dos 5 milhões.

Assim como para os usuários de *iOS*, esta categoria parece promissora também para os usuários de *Android*. No entanto, apenas desenvolver e lançar um aplicativo que sirva como biblioteca/*reader* não parece interessante, uma vez que já existem vários destes. Para cumprir o objetivo de também fugir de um mercado saturado e atrair usuários, seria preciso embutir funções diferentes, como por exemplo a opção de fóruns de discussão, ter um dicionário embutido para consultar palavras, uma versão dos livros em áudio e até *quizzes* em relação ao conteúdo dos livros. Pode-se também ter a possibilidade de criar listas de livros compartilhadas, para os quais os usuários possam descobrir novas possibilidades de leituras. O aplicativo poderia possuir apenas livros de domínio público, o que diminuiria a quantidade de conteúdo que teria de se criar em a partir dos livros, ou poderia focar em um nicho específico, como livros infantis ou livros de auto-ajuda, por exemplo - o que seria o objeto de uma outra análise.

### Conclusões

Neste projeto, analisamos dados provenientes de dois conjuntos diferentes, sendo eles de aplicativos para *Android* e de aplicativos de *iOS*. O objetivo era encontrar um perfil de aplicativo popular entre os usuários de ambos sistemas operacionais

A partir deste estudo foi possível concluir que o segmento de livros é um segmento popular para os usuários de ambas as plataformas. Desenvolver um aplicativo desta categoria pode ser interessante, contanto que se faça o esforço de criar novas *features*, como mencionado acima.